azerothcore/azerothcore-wotlk
# update (core): improve ChrRace DBC handling

Fixed up the ChrRace DBC Handling. Added addition Enums from TC Cherry Pick. The cherry pick came from someone who isn't a ass-clown. Testing was consistent with HelloKitty's Results. Alot of other adjustments to fit this alignment. Also our struct for ChrRace DBC was wrong and missing fields entirely.

Co-Authored-By: HelloKitty <5829095+HelloKitty@users.noreply.github.com>



## Changes Proposed:
Per HelloKitty's Template
```
Changes proposed:

Using ChrRace's Faction Template reference to calculate Player's Team instead of the base Language field in the ChrRace DBC
For the purposes of general use of TrinityCore for MMORPG development I think it's best not to tie the faction/alliance of a playable race entry to its base/common language. Instead, I think it'd be better to compute the faction/alliance association based on the actual faction template associated with the ChrRace.dbc entry.

TeamId in the ChrRaces.dbc structure in TrinityCore actually refers to the base/common language used by the race which links to Languages.dbc. Maybe that field should be renamed too.

Add Alliance type handling for ChrRaces.dbc to determine race playability
Add ChrRaceFlags to remove literal references
Cannot take any credit for the information. It's based on information from: https://wowdev.wiki/DB/ChrRaces
```

## Issues Addressed:

- Closes 

## SOURCE: 
Cherry pick of https://github.com/TrinityCore/TrinityCore/pull/24508


## Tests Performed:

- Consistent with HelloKitty's Tests
```
Logged in and debugged, correct calculation. Still aggressive/friendly with correct alliance/horde factions.
Created characters on both factions, still works.
```


## How to Test the Changes:


1. Logged in and\or debugged, and observe correct calculation still applied. 
2. Still aggressive/friendly with correct alliance/horde factions.
3. Created characters on both factions, still works.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
